packages/ui/src/components/va-tree-view/VaTreeView.demo.vue

```diff
@@ -1,59 +1,340 @@
+      <va-tree-view
+        :nodes="nodes"
+        :selection-type="selectionType"
+        selectable
+      />
+    </VbCard>
+    <VbCard title="Colored checkboxes">
+      <va-color-palette
+        v-model="selectedColor"
+        :palette="colorsPalette"
+      />
+      <va-tree-view
+        :nodes="nodes"
+        selectable
+        :color="selectedColor"
+      />
+    </VbCard>
+    <VbCard title="Selection model">
+      <p>Selection strategy</p>
+      <va-radio
+        v-for="(option, index) in selectionTypeOptions"
+        :key="index"
+        v-model="selectionType"
+        :option="option"
+      />
+      <p>
+        Single selection model for two tree views:
+        <br />
+        {{ JSON.stringify(selectedNodes) }}
+      </p>
+      <va-tree-view
+        v-model:checked="selectedNodes"
+        :nodes="nodes"
+        :selection-type="selectionType"
+        selectable
+      />
+      <va-tree-view
+        v-model:checked="selectedNodes"
+        :selection-type="selectionType"
+        :nodes="nodes"
+        selectable
+      />
+    </VbCard>
+    <VbCard title="valueBy">
+      <p>{{ JSON.stringify(selectedNodesValueBy) }}</p>
+      <br />
+      <va-tree-view
+        v-model:checked="selectedNodesValueBy"
+        :nodes="nodes"
+        :value-by="({id, label}) => `${id}_${label}`"
+        selectable
+      />
+    </VbCard>
+    <VbCard title="textBy">
+      <va-tree-view
+        :nodes="nodesTextBy"
+        text-by="title"
+      />
     </VbCard>
-    <VbCard title="Color">
-      <va-tree-view
-        color="info"
-      >
-        <va-tree-category
-          label="Category text"
-          is-open
-        >
-          <va-tree-node>
-            Node text
-          </va-tree-node>
-          <va-tree-node>
-            Node text
-          </va-tree-node>
-        </va-tree-category>
+    <VbCard title="trackBy">
+      <va-tree-view
+        :nodes="nodesTrackBy"
+        track-by="trackId"
+      />
+    </VbCard>
+    <VbCard title="iconBy">
+      <va-tree-view
+        :nodes="nodesIconBy"
+        icon-by="image"
+      />
+    </VbCard>
+    <VbCard title="disabledBy">
+      <va-tree-view
+        :nodes="nodesDisabledBy"
+        disabled-by="inactive"
+      />
+    </VbCard>
+    <VbCard title="Expandable">
+      <p>Expand all</p>
+      <va-tree-view :nodes="nodes" expand-all />
+      <p>Expanded by</p>
+      <va-tree-view :nodes="nodes" expand-all />
+      <p>Expanded v-model: {{ JSON.stringify(expandedNodes) }}</p>
+      <va-tree-view v-model:expanded="expandedNodes" :nodes="nodes" />
+    </VbCard>
+    <VbCard title="Filter">
+      <va-input
+        v-model="filterValue"
+        placeholder="Filter the tree view"
+        style="margin-bottom: 1rem;"
+      />
+      <va-checkbox v-model="isFilterCaseSensitive" label="Case sensitive" />
+      <va-tree-view
+        :nodes="filterableNodes"
+        :filter="filterValue"
+        :filter-method="customFilterMethod"
```

Comment:
Is there any point having both `filterValue` and `customFilterMethod`? Can't we have just one prop for both?